python/ray/train/tests/test_backend.py

```diff
@@ -106,6 +121,35 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: TestConfig):
         pass
 
 
+class MockWorkerGroup(WorkerGroup):
+    # Mock construct metadata
+    def add_workers(self, num_workers: int):
+        """Hard code the WorkerMetadata so that ther will be two nodes, on which
+        even and odd indexed workers are run respectively. For example if there are 4
+        workers, then worker 0 and 2 will be on node 0 and the rest on node 1.
+        """
+        new_actors = []
+        new_actor_metadata = []
+        for i in range(num_workers):
+            actor = self._remote_cls.options(
+                placement_group=self._placement_group
+            ).remote(*self._actor_cls_args, **self._actor_cls_kwargs)
+            new_actors.append(actor)
+            new_actor_metadata.append(
+                WorkerMetadata(
+                    node_id=str(i % 2),
+                    node_ip=str(i % 2),
+                    hostname=str(i % 2),
+                    gpu_ids=[str(i % 2)],
+                )
+            )
+
+        for i in range(len(new_actors)):
+            self.workers.append(
+                Worker(actor=new_actors[i], metadata=new_actor_metadata[i])
+            )
```

Comment:
I think we can simplify the amount we are overriding here and leverage super() for most of the logic:

```python
def add_workers(self, num_workers: int):
    super().add_workers(num_workers)
    for i, worker in enumerate(self.workers):
        metadata = WorkerMetadata(node_id=0,  node_ip=str(i % 2), hostname=0,  gpu_ids=[0])
        worker.metadata = metadata
```

Reply:
updated!